src/common/CryptoUtils.ts

```diff
@@ -132,6 +132,72 @@ export class CryptoUtils {
         };
     }
 
+    /**
+     * Converts a certificate from PEM to DER format.
+     *
+     * @param pemCertificate A string representing a certificate in PEM format.
+     * @returns A buffer containing the data of the certificate in DER format.
+     */
+    public static PEMtoDER(pemCertificate: string): Buffer {
+        const cert = this.getCertFromPEM(pemCertificate);
+        const derCertString = forge.asn1.toDer(forge.pki.certificateToAsn1(cert)).getBytes();
+        const derCertBuffer = Buffer.from(derCertString, 'binary');
+        return derCertBuffer;
+    }
+
+    /**
+     * Converts a certificate from DER to PEM format.
+     *
+     * @param derCertificate A buffer containing the binary data of the certificate in DER format.
+     * @returns A string representing a certificate in PEM format.
+     */
+    public static DERtoPEM(derCertificate: Buffer): string {
+        const cert = this.getCertFromDER(derCertificate);
+        const pemCert = forge.pki.certificateToPem(cert);
+        return pemCert;
+    }
+
+    /**
+     * Computes the subject hash (similar to subject_hash_old of OpenSSL). When invoking
+     * this method, the caller should either provide a DER or PEM certificate (but not both).
+     *
+     * @param derCertificate A buffer containing the data of the certificate in DER format.
+     * @param pemCertificate A string representing a certificate in PEM format.
+     * @returns A string representing the subject hash.
+     */
+    public static getSubjectHashOld(derCertificate?: Buffer, pemCertificate?: string): string {
```

Comment:
I could either have one function `getSubjectHashOld(derCertificate?: Buffer, pemCertificate?: string)` or break it up to two:
```
getDERSubjectHashOld(derCertificate: Buffer)
getPEMSubjectHashOld(pemCertificate: string)
```

Same for `isExpired()` function below. I don't have a particular preference for one over the other so will leave it up to the team to decide which approach to go with

Reply:
I prefer having two method signatures. It will be much clearer to the consumer what the desired inputs should be, to get their desired outputs. I'd recommend that for anywhere that we're currently saying that "you need to provide this input parameter _or_ that input parameter, but not both."

Reply:
Another way to do it (I don't have a preference) would be to have a data structure that encapsulates _all_ formats that we support, e.g. `Certificate`, that has members like `pemRepresentation`, `derRepresentation`, etc. It opens the door to encapsulating conversion between them, into that umbrella class. But I'd prefer one of those two approaches, over the either/or method parameters.